casper/src/main/scala/coop/rchain/casper/rholang/RuntimeManager.scala

```diff
@@ -285,4 +324,16 @@ object RuntimeManager {
       scodec.codecs.bytes,
       deployMergeableDataSeqCodec
     )
+
+  // TODO add more callbacks (system deploys, maybe something else)
+  trait ExecutionTracker[F[_]] {
```

Comment:
Why are `ExecutionTracker` and `noOpExecutionTracker` placed in the `RuntimeManager`? Is it his responsibility? Its responsibility is not clear from the name and set of methods. Many things for tests are initialized in `engine/Setup.scala`. Maybe move `noOpExecutionTracker` there?

Reply:
My logic was that runtime manager is responsible for controlling runtimes and executions done by these runtimes, and this trait is about tracking executions, providing callbacks that runtime manager can use to speak with outside world.